doc/source/how-to/testing.rst

```diff
@@ -0,0 +1,536 @@
+of the executable and usable source that are tested by unit tests. You can use
+the `pytest-cov <https://pytest-cov.readthedocs.io/>`_ library to view the
+coverage for your library.
+
+
+Configuring Code Coverage
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Code coverage must be properly configured, otherwise the resultant report will
+not be showing the real scope covered by the test suite.
+
+Assuming that a ``PyAnsys`` project follows :ref:`The \`\`src/\`\` Directory` layout,
+the following flag needsto be passed when :ref:`Running Tests`:
+
+.. code-block:: text
+
+    pytest --cov=ansys.<product>.<library> --cov-report=term tests/
+
+Previous command indicates ``pytest-cov`` to look for source code in the
+``src/ansys/<product>`` directory and generate a terminal report for all tests
+located in :ref:`The \`\`tests/\`\` Directory`.
+
+
+While 100% coverage is ideal, the law of diminishing returns often applies to
+the coverage of a Python library. Consequently, achieving 80-90% coverage is
+often sufficient.  For parts of your library that are difficult or impossible
+to test, consider using ``# pragma: no cover`` at the end of the method
+definition, branch, or line to denote that part of the code cannot be
+reasonably tested.  For example, if part of your module performs a simple
+``import`` test of ``matplotlib`` and raises an error when the library is not
+installed, it is not reasonable to attempt to test this and assume full
+coverage:
+
+.. code:: python
+
+   try:
+       import matplotlib
+   except ImportError:  # pragma: no cover
+       raise ImportError("Install matplotlib to use this feature.")
+
+You should only avoid coverage of parts of your library where you cannot
+reasonably test without an extensive testing suite or setup.  Most methods and
+classes, including edge cases, can be reasonably tested. Even parts of your code
+that raise errors like ``TypeError`` or ``ValueError`` when users input the
+wrong data type or value can be reasonably tested.
+
+Code Coverage Enforcement
+~~~~~~~~~~~~~~~~~~~~~~~~~
+One way of enforcing unit test coverage with a project on GitHub is to use the
+codecov.io to enforce minimum patch (and optionally project) coverage. As this
+application is already available to the `PyAnsys Organization
+<https://github.com/pyansys>`_, simply add a ``codecov.yml`` file to the root
+directory of your repository:
+
+.. code:: yaml
+
+   comment:
+     layout: "diff"
+     behavior: default
+
+   coverage:
+     status:
+       project:
+         default:
+           # basic
+           # target: 50%
+           threshold: 0%
+           # advanced
+           if_not_found: success
+           if_ci_failed: error
+           if_no_uploads: error
+       patch:
+         default:
+           # basic
+           target: 90%
+           if_not_found: success
+           if_ci_failed: error
+           if_no_uploads: error
+
+This requires that each PR has a patch coverage of 90%, meaning that 90% of any
```

Comment:
```suggestion
Using a ``codecov.yml`` file requires that each PR has a patch coverage of 90%, meaning that 90% of any
```

Reply:
I didn't know what "This" referred to so tried to replace the pronoun with a "real" subject. Feel free to change the subject  if needed.